src/Compilers/Core/Portable/SynthesizedLocalKind.cs

```diff
@@ -207,6 +204,11 @@ internal enum SynthesizedLocalKind
         InstrumentationPayload = 34,
 
         /// <summary>
+        /// Temp created for pattern matching by type.
+        /// </summary>
+        PatternMatchingTemp = 35,
```

Comment:
I'd remove "Temp" from the name. "Temp" suggests it's temporary, not long lived.
It would also be useful to mention in the comment what value is specifically stored in this variable. Is it the type object we are switching on?